atlasdb-commons/src/main/java/com/palantir/util/AtlasDbMetrics.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import com.codahale.metrics.Slf4jReporter;
+
+public class AtlasDbMetrics {
+
+    private static final String DEFAULT_LOG_DOMAIN = "com.palantir.util";
+    private static final String DEFAULT_REGISTRY_NAME = "AtlasDb";
+    private static final int DEFAULT_LOG_PERIOD_SECONDS = 60;
+    private static volatile AtlasDbMetrics instance = null;
+
+    private MetricRegistry metricRegistry;
+    private final Logger metricsLogger;
+    private final Slf4jReporter metricsLogReporter;
+
+    private AtlasDbMetrics(String logDomain, String registryName) {
+        this.metricRegistry = SharedMetricRegistries.getOrCreate(registryName);
```

Comment:
I want to be able to inject the MetricRegistry used, for example one from DropWizard's Environment#getMetrics()